app/utils/PopulateYear.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import play.api.libs.json.{JsValue, Json}
+
+import java.time.LocalDate
+
+object PopulateYear {
+  private val currentYear = "{{CURRENT-YEAR}}"
+  private val previousYear = "{{PREVIOUS-YEAR}}"
+
+  def apply(jsValue: JsValue): JsValue = {
+    val year = LocalDate.now().getYear
```

Comment:
@SiddharthAtos  - are you sure this would not cause any failures on the 1st Jan 2025 ?

Reply:
Let me adjust it with Tax Year.

Reply:
https://github.com/hmrc/income-tax-view-change-dynamic-stub/pull/91